tests/test_models.py

```diff
@@ -1,37 +1,192 @@
 
-    def test_example_replace_this(self):
-        """ It should always be true """
-        self.assertTrue(True)
+    # def test_example_replace_this(self):
+    #     """ It should always be true """
+    #     self.assertTrue(True)
+    def test_create_shopcarts(self):
+        """Test create shopcarts record"""
+        shop_cart = ShopCarts(customer_id=1, product_id=1, quantities=1)
+        shop_cart.create()
+        self.assertNotEqual(shop_cart, None)
+        self.assertEqual(shop_cart.customer_id, 1)
+        self.assertEqual(shop_cart.product_id, 1)
+        self.assertEqual(shop_cart.quantities, 1)
+        self.assertIsInstance(shop_cart.id, int)
+
+    def test_find_shopcarts(self):
+        """Test get a shopcarts record by its id"""
+        shop_cart1 = ShopCartsFactory()
+        shop_cart2 = ShopCartsFactory()
+        shop_cart3 = ShopCartsFactory()
+        shop_cart1.create()
+        shop_cart2.create()
+        shop_cart3.create()
+        uid = shop_cart2.id
+        shop_cart = ShopCarts.find(uid)
+        self.assertEqual(shop_cart2.id, shop_cart.id)
+        self.assertEqual(shop_cart2.customer_id, shop_cart.customer_id)
+        self.assertEqual(shop_cart2.quantities, shop_cart.quantities)
+
+    def test_update_shopcarts(self):
+        """Test update shopcarts record"""
+        shop_cart = ShopCartsFactory()
+        shop_cart.create()
+
+        shop_cart = ShopCarts.find(shop_cart.id)
+        shop_cart.customer_id = 15
+        shop_cart.product_id = 15
+        shop_cart.quantities = 15
+        shop_cart.update()
+
+        shop_cart = ShopCarts.find(shop_cart.id)
+        self.assertEqual(shop_cart.customer_id, 15)
+        self.assertEqual(shop_cart.product_id, 15)
+        self.assertEqual(shop_cart.quantities, 15)
+
+    def test_delete_shopcarts(self):
+        """Test delete shopcarts record by its id"""
+        shop_cart = ShopCartsFactory()
+        shop_cart.create()
+        uid = shop_cart.id
+        shop_cart.delete()
+
+        shop_cart = ShopCarts.find(uid)
+        self.assertIsNone(shop_cart)
+
+    def test_serialize_shopcarts(self):
+        """Test serialize shopcarts record"""
+        shop_cart = ShopCarts(customer_id=1, product_id=1, quantities=1)
+        shop_cart.create()
+        shop_cart_dict = shop_cart.serialize()
+
+        self.assertEqual(shop_cart_dict['customer_id'], 1)
+        self.assertEqual(shop_cart_dict['product_id'], 1)
+        self.assertEqual(shop_cart_dict['quantities'], 1)
+
+    def test_deserialize_shopcarts(self):
+        """Test deserialize shopcarts record"""
+        shop_cart_dict = {"id": 3, "customer_id": 1,
+                          "product_id": 1, "quantities": 1}
+        shop_cart = ShopCarts()
+        shop_cart.deserialize(shop_cart_dict)
+        self.assertEqual(shop_cart.customer_id, 1)
+        self.assertEqual(shop_cart.product_id, 1)
+        self.assertEqual(shop_cart.quantities, 1)
+
```

Comment:
I think you could also add test cases for deserialization missing data and bad data to improve code coverage

Reply:
That is work